extensions/blocks/amazon/edit.js

```diff
@@ -1,23 +1,261 @@
+		const selectedIds = selectedProducts.map( selectedProduct => {
+			const parsed = idRegex.exec( selectedProduct );
+			const selectedId = parsed[ 1 ] || parsed[ 2 ];
+			return data.products.filter( filteredProduct => filteredProduct.asin === selectedId );
+		} );
+		setAttributes( { asin: selectedIds[ 0 ][ 0 ].asin } );
+	};
+
+	const blockPlaceholder = (
+		<Placeholder
+			label={ __( 'Amazon', 'jetpack' ) }
+			instructions={ __( 'Search by entering an Amazon product name or ID below.', 'jetpack' ) }
+			icon={ <BlockIcon icon={ icon } /> }
+			notices={
+				notice && (
+					<Notice status="error" isDismissible={ false }>
+						{ notice }
+					</Notice>
+				)
+			}
+		>
+			<form>
+				<FormTokenField
+					value={ asin }
+					suggestions={ suggestions }
+					onInputChange={ onInputChange }
+					maxSuggestions={ 10 }
+					label={ __( 'Products', 'jetpack' ) }
+					onChange={ onChange }
+				/>
+				<Button isSecondary isLarge type="submit">
+					{ __( 'Preview', 'jetpack' ) }
+				</Button>
+			</form>
+		</Placeholder>
+	);
 
-export default AmazonEdit;
+	const inspectorControls = (
+		<InspectorControls>
+			{ asin && (
+				<>
+					<PanelBody title={ __( 'Promotion Settings', 'jetpack' ) }>
+						<ToggleControl
+							label={ __( 'Show Image', 'jetpack' ) }
+							checked={ showImage }
+							onChange={ () => setAttributes( { showImage: ! showImage } ) }
+						/>
+						<ToggleControl
+							label={ __( 'Show Title', 'jetpack' ) }
+							checked={ showTitle }
+							onChange={ () => setAttributes( { showTitle: ! showTitle } ) }
+						/>
+						<ToggleControl
+							label={ __( 'Show Author/Seller', 'jetpack' ) }
+							checked={ showSeller }
+							onChange={ () => setAttributes( { showSeller: ! showSeller } ) }
+						/>
+						<ToggleControl
+							label={ __( 'Show Price', 'jetpack' ) }
+							checked={ showPrice }
+							onChange={ () => setAttributes( { showPrice: ! showPrice } ) }
+						/>
+						<ToggleControl
+							label={ __( 'Show Purchase Button', 'jetpack' ) }
+							checked={ showPurchaseButton }
+							onChange={ () => setAttributes( { showPurchaseButton: ! showPurchaseButton } ) }
+						/>
+					</PanelBody>
+					<PanelColorSettings
+						title={ __( 'Color Settings', 'jetpack' ) }
+						colorSettings={ [
+							{
+								value: backgroundColor,
+								onChange: newBackgroundColor =>
+									setAttributes( { backgroundColor: newBackgroundColor } ),
+								label: __( 'Background Color', 'jetpack' ),
+							},
+							{
+								value: textColor,
+								onChange: newTextColor => setAttributes( { textColor: newTextColor } ),
+								label: __( 'Text Color', 'jetpack' ),
+							},
+							{
+								value: buttonAndLinkColor,
+								onChange: newButtonAndLinkColor =>
+									setAttributes( { buttonAndLinkColor: newButtonAndLinkColor } ),
+								label: __( 'Button & Link Color', 'jetpack' ),
+							},
+						] }
+					>
+						{
+							<ContrastChecker
```

Comment:
Is it standard to put this below the other controls? I had my dev tools open and missed the warning.

Reply:
That seems to be the way it's done in Gutenberg

Reply:
Interesting. I suppose it's stop the other controls moving around